lib/graceful-exit.js

```diff
@@ -148,8 +150,22 @@ exports.middleware = function middleware (app) {
 
   return function checkIfExitingGracefully (req, res, next) {
     if (app.settings.graceful_exit === true) {
-      // sorry keep-alive connections, but we need to part ways
-      req.connection.setTimeout(1);
+      if (options.respondDuringExit) {
+        logger('Server unavailable, incoming request rejected with error: ' + req && req.headers || '?');
```

Comment:
isn't `req.headers` an object? Won't this render as `Server unavailable, incoming request rejected with error: [object Object]`?

Reply:
+1

Reply:
Will do. Some loggers are smarter than that (what I use currently), but most won't be, including `console.log`.

Reply:
@CacheControl and @arp122 - pretty much rewrote the exit handler, including use of `util.inspect()` for the headers.